Microsoft.Toolkit.HighPerformance/Buffers/StringPool.cs

```diff
@@ -0,0 +1,775 @@
+
+            /// <summary>
+            /// The current number of items stored in the map.
+            /// </summary>
+            private int count;
+
+            /// <summary>
+            /// The 1-based index for the start of the free list within <see cref="mapEntries"/>.
+            /// </summary>
+            private int freeList;
+
+            /// <summary>
+            /// The current incremental timestamp for the items stored in <see cref="heapEntries"/>.
+            /// </summary>
+            private uint timestamp;
+
+            /// <summary>
+            /// A type representing a map entry, ie. a node in a given list.
+            /// </summary>
+            private struct MapEntry
+            {
+                /// <summary>
+                /// The precomputed hashcode for <see cref="Value"/>.
+                /// </summary>
+                public int HashCode;
+
+                /// <summary>
+                /// The <see cref="string"/> instance cached in this entry.
+                /// </summary>
+                public string Value;
+
+                /// <summary>
+                /// The 0-based index for the next node in the current list.
+                /// </summary>
+                public int NextIndex;
+
+                /// <summary>
+                /// The 0-based index for the heap entry corresponding to the current node.
+                /// </summary>
+                public int HeapIndex;
+            }
+
+            /// <summary>
+            /// A type representing a heap entry, used to associate priority to each item.
+            /// </summary>
+            private struct HeapEntry
+            {
+                /// <summary>
+                /// The timestamp for the current entry (ie. the priority for the item).
+                /// </summary>
+                public uint Timestamp;
+
+                /// <summary>
+                /// The 0-based index for the map entry corresponding to the current item.
+                /// </summary>
+                public int MapIndex;
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="FixedSizePriorityMap"/> struct.
+            /// </summary>
+            /// <param name="capacity">The fixed capacity of the current map.</param>
+            public FixedSizePriorityMap(int capacity)
+            {
+                this.buckets = new int[capacity];
+                this.mapEntries = new MapEntry[capacity];
+                this.heapEntries = new HeapEntry[capacity];
+                this.count = 0;
+                this.freeList = EndOfList;
+                this.timestamp = 0;
+            }
+
+            /// <summary>
+            /// Gets an <see cref="object"/> that can be used to synchronize access to the current instance.
+            /// </summary>
+            public object SyncRoot
+            {
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
i assume the attribute here is just for intent? as it is useless

Reply:
Yeah, I always like to be explicit in all these small properties, especially in performance focused types. At worst, the attribute doesn't hurt and just changes nothing anyway, but I really like the extra clarity 😊